prov/verbs/src/fi_verbs.c

```diff
@@ -2150,50 +2146,40 @@ static int fi_ibv_copy_addr(void *dst_addr, size_t *dst_addrlen, void *src_addr)
 static int fi_ibv_msg_ep_setname(fid_t ep_fid, void *addr, size_t addrlen)
 {
 	struct fi_ibv_msg_ep *ep;
-	struct fi_info info;
+	void *addr_temp;
 	struct rdma_cm_id *id;
 	int ret;
 
 	ep = container_of(ep_fid, struct fi_ibv_msg_ep, ep_fid);
 
-	if (addrlen != ep->src_addrlen) {
+	if (addrlen != ep->info->src_addrlen) {
 		FI_INFO(&fi_ibv_prov, FI_LOG_EP_CTRL,"addrlen expected: %d, got: %d.\n",
-				ep->src_addrlen, addrlen);
+				ep->info->src_addrlen, addrlen);
 		return -FI_EINVAL;
 	}
 
-	memset(&info, 0, sizeof info);
-	info.src_addr = calloc(1, addrlen);
-	if (!info.src_addr)
+	addr_temp = calloc(1, ep->info->src_addrlen);
+	if (!addr_temp)
 		return -FI_ENOMEM;
 
-	memcpy(info.src_addr, addr, addrlen);
-	info.src_addrlen = addrlen;
+	memcpy(addr_temp, ep->info->src_addr, ep->info->src_addrlen);
 
-	if (ep->dst_addr) {
-		info.dest_addrlen = ep->dst_addrlen;
-		info.dest_addr = calloc(1, info.dest_addrlen);
-		if (!info.dest_addr) {
-			ret = -FI_ENOMEM;
-			goto err1;
-		}
+	memcpy(ep->info->src_addr, addr, ep->info->src_addrlen);
```

Comment:
Got it.  Then I would save off the src_addr pointer, versus copying src_addr -- minor optimization.  Otherwise, rest looks good to me.
save_addr = src_addr;
src_addr = malloc(..)
memcpy(src_addr, addr, ...)
free(save_addr)